libs/vkd3d/swapchain.c

```diff
@@ -2421,23 +2416,25 @@ static HRESULT d3d12_swapchain_init(struct d3d12_swapchain *swapchain, IDXGIFact
         }
     }
 
-    IDXGIFactory_AddRef(swapchain->factory = factory);
-
-    if (FAILED(hr = d3d12_swapchain_set_fullscreen(swapchain, target, TRUE)))
+    if (FAILED(d3d12_swapchain_set_fullscreen(swapchain, target, TRUE)))
     {
-        WARN("Failed to go fullscreen.\n");
+        ERR("Failed to enter fullscreen.");
         return hr;
     }
 
-    if (!output && target)
+    if (!swapchain->fullscreen_desc.Windowed)
+        swapchain->target = target;
```

Comment:
Do we initialize swapchain->target anywhere if we don't take this code path?

Reply:
It is set to 0 already given it is calloc'ed